app/src/main/java/com/sdp13epfl2021/projmag/database/UserDataFirebase.kt

```diff
@@ -129,4 +138,35 @@ class UserDataFirebase(
                 .addOnFailureListener(onFailure)
         }
     }
+
+    override fun applyUnapply(
+        apply: Boolean,
+        projectId: ProjectId,
+        onSuccess: () -> Unit,
+        onFailure: (Exception) -> Unit
+    ) {
+        getUserDoc()?.let { doc ->
+            val fieldValue = if (apply) {
+                FieldValue.arrayUnion(projectId)
+            } else {
+                FieldValue.arrayRemove(projectId)
+            }
+            doc.update(APPLIED_TO_FIELD, fieldValue)
+                .addOnSuccessListener { onSuccess() }
+                .addOnFailureListener(onFailure)
```

Comment:
If the /user-data/{userID} document does not exists, the applied list is not updated.
You can either try to create it if it does not exists, or we can also rely on the fact that the document will be created after the user register.